### DISCUSSION TITLE
Result<Result<T>> and the absence of Bind

### DISCUSSION THREAD

**Discussion by megasuperlexa** [louthy/language-ext#1312] [Q&A] [answered]
Is there any reason to not have `Bind` for `Result`?
I am now ended up writing numerous `Match (success => new Result(TransfromSuccessToResultOfAnotherT()), err => new Result<AnotherT>(err))` methods instead of simply Binding my transformation.

**Comment by aloslider** [comment]
You need to bind `Try`s, not `Result`s:  
```cs
Try<T> tryT = () => default(T);
Func<T, Try<AnotherT>> tryAnotherT = (T t) => () => MapTToAnotherT(t);
Try<AnotherT> operation = tryT.Bind(tryAnotherT);

**Comment by StefanBertels** [comment, marked as answer]
This might be helpful, too:

https://github.com/louthy/language-ext/discussions/1226

**Comment by megasuperlexa** [comment]
> This might be helpful, too:
> 
> #1226


Thanks. I wish it was more articulated, e.g. via the`[Obsolete]` attribute.

**Comment by louthy** [comment]
Because of the endless confusion about `Result`, I have removed it from `v5.0.0` -- `Fin` is recommended (as has already been stated by other commenters)
